src/pages/create-issue/index.jsx

```diff
@@ -38,25 +42,51 @@ const styles = theme => ({
   },
 });
 
+function writeStateToStorage(state) {
+  return store('issueistFormData', {
+    selectedRepository: state.selectedRepository,
+    title: state.title,
+    body: state.body
+  })
+}
+
+function retrieveStateFromStorage() {
+  return retrieve('issueistFormData')
+}
+
 class App extends Component {
   state = {
     repositories: [],
     selectedRepository: '',
     title: '',
     body: '',
+    timer: '',
     error: null,
     loading: false,
   }
 
-  handleChange = (event) => {
-    this.setState({ [event.target.name]: event.target.value });
+  componentDidMount() {
+    // First update the component's state with any todo data that was
+    // persisted in the store.
+    retrieveStateFromStorage()
+      .then((state) => this.setState({...state}))
+
+    // basic auth
+    this.gh = new GitHub({
```

Comment:
Should this be done here or passed down through components?

Reply:
I'm not 100% sure. One part of me thinks that it'd be nice if this component were instantiated with the starting state (if it exists), but I also feel like the starting state is very specific to the component and because of that it's fine if the component knows how to load it/save it itself.

If you feel strongly about passing it in via props, I'd be fine with that, but we should also move the state-saving part into a function passed via props as well. Thoughts?